src/main/webapp/app/iris/manage/settings/shared/iris-enabled/iris-enabled.component.spec.ts

```diff
@@ -83,14 +86,120 @@ describe('IrisEnabledComponent', () => {
 
     it('should set course enabled', async () => {
         const setSettingsSpy = jest.spyOn(irisSettingsService, 'setCourseSettings').mockReturnValue(of(new HttpResponse({ body: null as any as IrisSettings })));
-        comp.course = course;
+        componentRef.setInput('course', course);
         comp.irisSettings = irisSettings;
-        comp.irisSubSettingsType = IrisSubSettingsType.CHAT;
+        componentRef.setInput('irisSubSettingsType', IrisSubSettingsType.CHAT);
         comp.irisSubSettings = irisSettings.irisChatSettings;
 
         comp.setEnabled(true);
         expect(setSettingsSpy).toHaveBeenCalledOnce();
         await Promise.resolve();
         expect(comp.irisSubSettings?.enabled).toBeTrue();
     });
+    it('should set all subsettings to enabled when IrisSubSettingsType is ALL and setEnabled is called with true', async () => {
+        const setSettingsSpy = jest.spyOn(irisSettingsService, 'setCourseSettings').mockReturnValue(of(new HttpResponse({ body: null as any as IrisSettings })));
+        jest.spyOn(irisSettingsService, 'getUncombinedCourseSettings').mockReturnValue(of({} as IrisCourseSettings));
+        componentRef.setInput('course', course);
+        componentRef.setInput('irisSubSettingsType', IrisSubSettingsType.ALL);
+        comp.irisSettings = {
+            ...irisSettings,
+            irisChatSettings: { type: IrisSubSettingsType.CHAT, enabled: false },
+            irisTextExerciseChatSettings: { type: IrisSubSettingsType.TEXT_EXERCISE_CHAT, enabled: false },
+            irisCourseChatSettings: { type: IrisSubSettingsType.COURSE_CHAT, enabled: false },
+            irisCompetencyGenerationSettings: { type: IrisSubSettingsType.COMPETENCY_GENERATION, enabled: false },
+            irisLectureChatSettings: { type: IrisSubSettingsType.LECTURE, enabled: false },
+            irisFaqIngestionSettings: { type: IrisSubSettingsType.FAQ_INGESTION, enabled: false, autoIngestOnFaqCreation: false },
+            irisLectureIngestionSettings: { type: IrisSubSettingsType.LECTURE_INGESTION, enabled: false, autoIngestOnLectureAttachmentUpload: false },
+        };
+        fixture.detectChanges();
+
+        comp.setEnabled(true);
+
+        expect(setSettingsSpy).toHaveBeenCalledOnce();
+        expect(comp.irisSettings.irisChatSettings?.enabled).toBeTrue();
+        expect(comp.irisSettings.irisTextExerciseChatSettings?.enabled).toBeTrue();
+        expect(comp.irisSettings.irisCourseChatSettings?.enabled).toBeTrue();
+        expect(comp.irisSettings.irisCompetencyGenerationSettings?.enabled).toBeTrue();
+        expect(comp.irisSettings.irisLectureChatSettings?.enabled).toBeTrue();
+        expect(comp.irisSettings.irisFaqIngestionSettings?.enabled).toBeTrue();
+        expect(comp.irisSettings.irisLectureIngestionSettings?.enabled).toBeTrue();
```

Comment:
I suggest we extract this block which is repeating for true and false below into a helper to avoid repetition